api/src/main/java/org/apache/iceberg/expressions/BoundAggregate.java

```diff
@@ -18,18 +18,40 @@
  */
 package org.apache.iceberg.expressions;
 
+import java.util.Map;
+import org.apache.iceberg.DataFile;
 import org.apache.iceberg.StructLike;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
 public class BoundAggregate<T, C> extends Aggregate<BoundTerm<T>> implements Bound<C> {
+
+  // indicates if there are enough stats for this aggregate to be calculated at
+  // iceberg side.
+  private boolean canPushDown = false;
```

Comment:
Aggregate expressions can't be mutable. The aggregator is the mutable state that gets created for every aggregation.